src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -14,17 +36,25 @@ export const TodoFilter = () => (
       <input
         data-cy="searchInput"
         type="text"
+        value={query}
         className="input"
         placeholder="Search..."
+        onChange={event => onQueryChange(event.target.value)}
       />
       <span className="icon is-left">
         <i className="fas fa-magnifying-glass" />
       </span>
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
+      {query && (
+        <span className={classNames('icon is-right', { pointerEvents: 'all' })}>
```

Comment:
You use the classnames library incorrectly, https://www.npmjs.com/package/classnames